episodes/3-identify-the-problem.Rmd

```diff
@@ -0,0 +1,301 @@
+
+
+```{r}
+krats
+```
+It's empty! What went wrong with our "Dipadomys" filter?
+
+```{r}
+rodents %>% count(genus)
+```
+We see two things here. For one, we've misspelled Dipodomys, which we can now amend. This quick function call also tells us we should expect a data frame with 9573 values resulting after subsetting to the genus Dipodomys.
+
+```{r}
+krats <- rodents %>% filter(genus == "Dipodomys") #kangaroo rat genus
+dim(krats)
+
+ggplot(krats, aes(year, fill=plot_type)) + 
+geom_histogram() +
+facet_wrap(~species)
+
+```
+Our improved code here looks good. A quick "dim" call confirms we now have all the Dipodomys observations, and our plot is looking better. In general, having a 'print' statement or some other output before plots or other major steps can be a good way to check your code is producing intermediate results consistent with your expectations. 
+
+However, there's something funky happening here. The bins are definitely weirdly spaced -- we can see some bins are not filled with any observations, while those exactly containing one of the integer years happens to contain all the observations for that year. 
+
+:::::::: challenge 
+As a group, name some potential problems or undesired outcomes from this graph...
+::::::::
+
+:::::::: solution
+
+ - The graph looks sparse, and unevenly so -- many bins have no observations
+ - Suggests that some years had more observations and others fewer based on somewhat arbitrary measurements (i.e. what calendar year happened to fall on)
+ - Hard to compare trends across time, or even subsequent years...
+
+::::::::
+
+As we discussed in the challenge, there are some issues to visualizing our data this way. A solution here might be to tinker with the bin width in the histogram code, but let's step back a minute. Do we necessarily need to dive into the depths of tinkering with the plot? We can evalulate this problem not in terms of the plot having a problem, but with our data type having a problem. There's  an opportunity to encode the observation times outside of coarse, somewhat arbitrary year groupings with the real, interpretable date they were collected. Let's do that using the tidyverse's 'lubridate' package. The important details here are that we are creating a 'datetime'-type variable using the recorded years, months, and days, which are currently all encoded as numeric types.
+
+```{r}
+krats <- rodents %>% filter(genus == "Dipodomys") #kangaroo rat genus
+dim(krats)
+
+krats <- krats %>% mutate(date = lubridate::ymd(paste(year,month,day,sep='-')))
+
+ggplot(krats, aes(date, fill=plot_type)) + 
+geom_histogram() +
+facet_wrap(~species)
+
+```
+
+This looks much better, and is easier to see the trends over time as well. Note our x axis still shows bins with year labelings, but the continuous spread of our data over these bins shows that dates are treated more continuously and fall more continuously within histogram bins.
+
+:::::::: callout 
```

Comment:
In general I think this is a good callout, but I'm not entirely following the connection to reproducible examples. Can you explain what you meant by that? Why is this example more relevant to reproducible examples than the other examples in this episode?